src/ArmTemplates/Common/Exceptions/DuplicateTagResourceNameException.cs

```diff
@@ -0,0 +1,16 @@
+﻿// --------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+// --------------------------------------------------------------------------
+
+using System;
+
+namespace Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.Exceptions
+{
+    public class DuplicateTagResourceNameException : Exception
+    {
+        public DuplicateTagResourceNameException(string message) : base(message)
```

Comment:
you can encapsulate logic of `string.Format` here. Like this:
```suggestion
         const string DuplicateTagResourceNameErrorMessageFormat = "Duplicate tag resource name found during sanitizing the display name. Please consider renaming tags: {0}, {1}. Both resulted resource name to be equal to: {2}";
 
        public DuplicateTagResourceNameException(string existingValue, string tagName, string resourceName) 
            : base(string.Format(ErrorMessages.DuplicateTagResourceNameErrorMessage, existingValue, tagName, resourceName))
```

My points for that implementation:
- I do not think `DuplicateTagResourceNameErrorMessageFormat` is used in any other exception, so we can easily hide it in corresponding exception
- usage of this exception becomes much more easy to understand - you tell developer explicitly what to pass to show in exception
```
throw new DuplicateTagResourceNameException(existingValue, tagName, resourceName);
```

Reply:
I was thinking about this approach but decided to go with this one to have all error messages in one place but not in separate classes per error. What do you think?

Reply:
As was discussed updated the exceptions classes with custom constructors per each introduced exception